### PR TITLE
Output ZIP files instead of XPI files (fix #325)

### LINKED ISSUE #325: Build to zip files by default
Since [AMO supports zip files](https://github.com/mozilla/addons-server/issues/2727) as well as Firefox (as temporary installs) and Chrome, we might as well build to zip as the default rather than `.xpi`.

### PR DESCRIPTION
**Pull request by tofumatt**
Fixes #325

r?

I tested signing and it still a. works with the zip (as expected) and b. sends down an XPI, which is expected behaviour.

Tested `web-ext run` and everything still worked as normal too.

### CODE REVIEW

**Comment by tofumatt** [comment]
I think this should also close #119, as it will package a ZIP file Firefox will load and (presumably) Chrome will load/accept.

**Comment by tofumatt** [comment]
Just uploaded the ZIP file `web-ext build` output and the Chrome Store accepted the upload and it worked, so I'd say #119 is closed by this, yup.

screenshot 2016-06-29 13 29 38

**Comment by kumar303** [comment]
Cool! I'm taking a look now. 

One thing we've been doing is making each commit message [auto-changelog-ready](https://github.com/mozilla/web-ext/blob/master/CONTRIBUTING.md#writing-commit-messages) so could you rename the message to "feat: Output ZIP files instead of XPI files" ? 

Also, the commit message itself should not mention the github issue (only the pull request should) otherwise every time you rebase and push, it will spam the issue thread (which gets messy).

**Comment by tofumatt** [comment]
Ten four, I shall fix these things.

**Comment by tofumatt** [comment]
Fixed both things!

File: src/cmd/build.js [hunk 1]
```diff
@@ -31,8 +31,9 @@ function defaultPackageCreator(
           filter: (...args) => fileFilter.wantFile(...args),
         })
         .then((buffer) => {
+          let extension = 'zip';
           let packageName = safeFileName(
-            `${manifestData.name}-${manifestData.version}.xpi`);
+            `${manifestData.name}-${manifestData.version}.${extension}`);
```
Comment (kumar303) [inline src/cmd/build.js]: might as well just make this `${manifestData.name}-${manifestData.version}.zip` -- no need for a variable

Comment (tofumatt) [inline src/cmd/build.js]: Oh yeah, you’re right. I was thinking of making the output extension an
option but then realised it’s unneeded. Shall remove.

File: tests/test-cmd/test.build.js [hunk 2]
```diff
@@ -209,7 +209,7 @@ describe('build', () => {
       assert.equal(filter.wantFile('some.xpi'), true);
       assert.equal(filter.wantFile('manifest.json'), false);
     });
-    
+
```
Comment (kumar303) [inline tests/test-cmd/test.build.js]: heh. I fixed the eslint rule for this in another PR that hasn't landed yet. I was like wut, how did that even get committed? 🎯

Comment (tofumatt) [inline tests/test-cmd/test.build.js]: And my editor just deletes trailing whitespace so I didn't even notice I committed it 😄

**Comment by kumar303** [comment]
> the Chrome Store accepted the upload

highfive

**Comment by kumar303** [comment]
At first I was wondering if we should make an option to output to `.xpi` if desired but I can't really think of any reason to :)

Looks good!

File: tests/test-cmd/test.build.js (hunk 2 above)
Comment (kumar303) [inline tests/test-cmd/test.build.js]: no, your change is correct. Thanks.

**Comment by kumar303** [comment]
oh, heh, I was waiting for fixing up the [extraneous variable](https://github.com/mozilla/web-ext/pull/328/files#r68977634). All good though, no need for an additional patch. Thanks for getting this in.

**Comment by tofumatt** [comment]
Shoot! I made the commit and didn't push it. I am dumb.

I'll make another patch, I hate silly extra lines.

**Comment by kumar303** [comment]
ok, if you do, prefix the commit message with `chore:` so that it gets excluded from the auto-changelog. It's not exactly a chore but hey.

**Comment by tofumatt** [comment]
Noted! Thanks :-)

**Pull request merged**